recipes-core/initrdscripts/tegra-flash-init_1.0.bb

```diff
@@ -14,6 +14,14 @@ UNPACKDIR = "${S}"
 
 COMPATIBLE_MACHINE = "(tegra)"
 
+OTABOOTDEV ?= "/dev/mtdblock0"
+
+do_configure() {
+    MTDBLOCK_DEV=$(basename ${OTABOOTDEV})
+    MTD_DEV_NUM="${MTDBLOCK_DEV##mtdblock}"
+    echo "MTD_DEV=mtd${MTD_DEV_NUM}" > ${WORKDIR}/tegra-flash.env
```

Comment:
Instead of deferring the substition to runtime, I'd suggest just setting MTD_DEV as a bitbake variable (e.g. `MTD_DEV ?= "${@d.getVar('OTABOOTDEV').replace('mtdblock','mtd')}"`),  and using a token like `@MTD_DEV@` in the script itself (in place of the `/dev/${MTD_DEV}` you have now).
 
Then add a `sed -i` command in the `do_install` step to replace the token with the variable's value.

That's the more typical pattern we follow in other recipes for handling values known at build time (and is more typical across other Yocto layers generally).

Then you don't need to install the extra `tegra-flash.env` file.